### PR TITLE
[3.2.0] Restructuring the doc on JWT grant with WSO2 IS as IDP

### PR DESCRIPTION
**Pull request by wasuradananjith**
## Purpose
Fixes: https://github.com/wso2/docs-apim/issues/1622

## Goals
Validated the steps and add missing content to set it up with WSO2 IS.

## Approach
- The page **JWT Grant** was updated as follows.
  image

### CODE REVIEW

File: en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md [hunk 1]
```diff
@@ -1,69 +1,137 @@
...
+You can use any identity provider to obtain a JWT. As an example, this step will discuss how to obtain a JWT from WSO2 Identity Server (WSO2 IS).
 
-    !!! info
-        Identity provider Public Certificate
+1. Download and install the [latest WSO2 Identity Server](https://wso2.com/identity-and-access-management/#).
+     
+     If you downloaded the archive, extract it. `<IS_HOME>` refers to the root folder of the extracted WSO2 IS.
 
-        The **Identity Provider Public Certificate** is the public certificate belonging to the identity provider. Uploading this is necessary to authenticate the response from the identity provider.
+     Start WSO2 Identity Server with a port offset. (`portOffset` is required only if you are running both API-M and ID in the same) JVM.
```
Comment (tgtshanika) [inline en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md:22]: Parenthesis has to be after JVM?

File: en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md [hunk 2]
```diff
@@ -1,69 +1,137 @@
...
+3. Navigate to the **Identity Providers** section in the **Main** -> **Identity** section and click **Add.**
+
+4. Provide the following values to configure the Identity Provider.
+    -   **Identity Provider Name :** Enter the issuer name as the identity provider name. This is used to generate the JWT assertion. The issuer of a JWT generated by WSO2 IS is by default `https://localhost:9444/oauth2/token`.
+    -   **Identity Provider Public Certificate :** Choose the radio button named **Upload IDP certificate** so that the field named **Identity Provider Public Certificate** will be enabled. This is the certificate which will be used to sign the JWT assertion.
+
+        !!! info
+            The **Identity Provider Public Certificate** is the public certificate belonging to the identity provider. Uploading this is necessary to authenticate the response from the identity provider.
+
+            This can be any certificate. If the identity provider is another API Manager or an Identity Server, this can be a `wso2.crt` file.
```
Comment (tgtshanika) [inline en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md:86]: cert file name can be anything

File: en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md [hunk 3]
```diff
@@ -1,69 +1,137 @@
...
+
+            - Open your command line interface, navigate to the `<IS_HOME>/repository/resources/security/` directory and run the following command.
+              ``` java
+              keytool -export -alias wso2carbon -file wso2.crt -keystore wso2carbon.jks -storepass wso2carbon
+              ```
 
-7.  Enter a **Callback URL.** For example, use `https://wso2is.local:8080/playground2/oauth2client` and click **Add**. The **OAuth Client Key** and **OAuth Client Secret** will now be visible.
+            - Once you run this command, the `wso2.crt` file will be generated inside the same folder. You can upload it by clicking **Choose File** in Identity Provider section in the Management Console.
+
+            See [Using Asymmetric Encryption](https://docs.wso2.com/display/ADMIN44x/Using+Asymmetric+Encryption) in the WSO2 Product Administration Guide for more information on how public keys work and how to sign these keys by a certification authority.
```
Comment (tgtshanika) [inline en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md:97]: This doc link is deprecated. Shall we point to the https://apim.docs.wso2.com/en/latest/install-and-setup/setup/security/configuring-keystores/keystore-basics/about-asymetric-cryptography/

File: en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md [hunk 4]
```diff
@@ -1,69 +1,137 @@
...
+6. Expand the **OAuth/OpenID Connect Configuration** under Inbound Authentication Configuration and click **Configure.**
+
+7. Enter a **Callback URL.** For example, use `http://localhost:8080/playground` and click **Add**. The **OAuth Client Key** and **OAuth Client Secret** will now be visible.
 
     [![]({{base_path}}/assets/img/learn/add-service-provider.png)]({{base_path}}/assets/img/learn/add-service-provider.png)
 
-### Using the JWT grant
+Now we have a registered identity provider and as well as a service provider with an **OAuth Client Key** and an **OAuth Client Secret**. In the next section we can test the JWT grant with the configured setup.
+
+## Using the JWT grant
```
Comment (tgtshanika) [inline en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md:120]: The same topic repeated twice.

File: en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md (hunk 1 above)
Comment (wasuradananjith) [reply en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md:22]: Fixed via https://github.com/wso2/docs-apim/pull/3170/commits/d99c31c64db63b3df9a1aad0bc9edde2c10379f0

File: en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md (hunk 2 above)
Comment (wasuradananjith) [reply en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md]: Fixed via https://github.com/wso2/docs-apim/pull/3170/commits/d99c31c64db63b3df9a1aad0bc9edde2c10379f0

File: en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md (hunk 3 above)
Comment (wasuradananjith) [reply en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md]: Fixed via https://github.com/wso2/docs-apim/pull/3170/commits/d99c31c64db63b3df9a1aad0bc9edde2c10379f0

File: en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md (hunk 4 above)
Comment (wasuradananjith) [reply en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md:103]: Restructured the steps via https://github.com/wso2/docs-apim/pull/3170/commits/938074070570a12605876265c41b0f28362e63d2

**Review by tgtshanika — APPROVED** [review]

**Review by Mariangela — APPROVED** [review]

**Pull request merged**